connectors/cli.py

```diff
@@ -91,17 +92,29 @@ def _parser():
 def run(args):
     """Runner"""
     # just display the list of connectors
+    config = EnvYAML(args.config_file)
     if args.action == "list":
         logger.info("Registered connectors:")
-        config = EnvYAML(args.config_file)
         for source in get_data_sources(config):
             logger.info(f"- {source.__doc__.strip()}")
         logger.info("Bye")
         return 0
 
+    loop = get_event_loop(args.uvloop)
+    preflight_check = PreflightCheck(config)
+    preflight_check_coro = preflight_check.run()
+
+    for sig in (signal.SIGINT, signal.SIGTERM):
+        loop.add_signal_handler(sig, functools.partial(preflight_check.shutdown, sig))
+
+    try:
+        if not loop.run_until_complete(preflight_check_coro):
```

Comment:
you cannot call `loop.run_until_complete` twice.

The proper pattern here is something like this:

```
async def start():
    preflight = PreflightCheck(config)
    checked = await preflight.run()
    if not checked:
        raise Something()
    service = SyncService(args)
    for sig in (signal.SIGINT, signal.SIGTERM):
        loop.add_signal_handler(sig, functools.partial(service.shutdown, sig))
    await service.run()


loop.run_until_complete(start_service())

```




```

Reply:
I made the change here: https://github.com/elastic/connectors-python/pull/297/commits/c729a55f5047fb35fbd5fcb2ac6856b77ed6bde1, with an additional signal handle for preflight check:
https://github.com/elastic/connectors-python/blob/c729a55f5047fb35fbd5fcb2ac6856b77ed6bde1/connectors/cli.py#L93-L94

It behaves similarly when testing locally, which can exit gracefully for both preflight check and sync service.

But when running test `test_main_and_kill`, it killed the test process, not the connector service:
```
connectors/tests/test_cli.py::test_main_and_kill make: *** [Makefile:48: test] Terminated
🚨 Error: The command exited with status 2
user command error: exit status 2
```